pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1689,7 +1689,14 @@ def _check_unnecessary_dict_index_lookup(
                         and subscript == subscript.parent.target
                     ):
                         # Ignore this subscript if it is the target of an assignment
-                        continue
+                        if subscript.as_string() == subscript.parent.value.as_string():
+                            # Fire error as d[k] += d[k] has an unnecessary index lookup
+                            self.add_message(
+                                "unnecessary-dict-index-lookup",
+                                node=subscript,
+                                args=(node.target.elts[1].as_string()),
+                            )
```

Comment:
I'm not really sure this is worth it. Keep in mind that every special case we add has the potential to introduce new errors and complicates the code further. For common cases that is acceptable, but in this instance I don't think it's beneficial.

Reply:
I was trying to preserve the original behavior of the code, 

from ``unnecessary_dict_index_lookup.py``, Line 37,

```python
Foo.c_dict[k] += Foo.c_dict[k]  # [unnecessary-dict-index-lookup]
```

Since with ``return``, the LHS is evaluated first and returns,  the RHS is thus not evaluated -- which led to me throwing this in to keep the functionality as similar as possible.

I'm okay with removing this, but I'm kinda concerned of nitpicks which may result in a false-negative issue for such instances in the near future.

Reply:
I would argue that it's okay because it's a rare case. We don't need to catch everything for a refactoring suggestion. As long as something like this still works, I believe it's fine
```py
# unnecessary_dict_index_lookup.py -> L37

    val = 0
    val += Foo.c_dict[k]  # [unnecessary-dict-index-lookup]
```

Reply:
That' sounds reasonable, let me remove it. Thanks for your input on this :)
```suggestion
```